tensorflow/core/kernels/data/name_utils.h

```diff
@@ -27,39 +29,65 @@ namespace name_utils {
 extern const char kDelimiter[];
 extern const char kDefaultDatasetDebugStringPrefix[];
 
+struct OpNameParams {
+  OpNameParams() = default;
+
+  explicit OpNameParams(int op_version) : op_version(op_version){};
+
+  int op_version = 1;
+};
+
+struct DatasetDebugStringParams {
+  DatasetDebugStringParams() = default;
+
+  template <typename... T>
+  explicit DatasetDebugStringParams(int op_version, string dataset_prefix,
+                                    const T&... args)
+      : op_version(op_version),
+        dataset_prefix(std::move(dataset_prefix)),
+        args({static_cast<const strings::AlphaNum&>(args).data()...}){};
+
+  int op_version = 1;
+  string dataset_prefix = "";
+  std::vector<string> args;
```

Comment:
`AlphaNum` is not used here because it is not allowed to copy and assign (`TF_DISALLOW_COPY_AND_ASSIGN(AlphaNum);`)